sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -53,6 +55,18 @@ class FileStreamSource(
     fs.makeQualified(new Path(path))  // can contain glob patterns
   }
 
+  private val sourceCleaner: FileStreamSourceCleaner = {
```

Comment:
I wasn't going to ask, but since I have more comments... I think it's better if this were an `Option` and set to `None` if the cleaner is off.

Similarly, below, you'll resolve the `sourceArchiveDir` even if the cleaner is not set to `archive`, which is not necessary.

(I'm almost suggesting that there should be a separate implementation for `delete` and for `archive` to make this, and the code calling it, a bit cleaner.)

Reply:
Yeah, that's a good suggestion. Will address.